### DISCUSSION TITLE
Should the trigger function from useLazyXxxQuery be included the useEffect dependency array?

### DISCUSSION THREAD

**Discussion by ensconced** [reduxjs/redux-toolkit#3458] [Q&A]
I have something like this:

```
  const [trigger, { data, isLoading }] = useLazyXxxQuery();

  useEffect(() => {
    trigger(arg, false);
  }, [arg]);
```

Should `trigger` be included in the dependency array? Is it safe to assume that it won't change and cause the effect to fire more often than intended?

**Comment by EskiMojo14** [comment]
what you're doing doesn't seem functionally different to 
```ts
const { data, isLoading } = useXxxQuery(arg, { refetchOnMountOrArgChange: true })
```

i would recommend doing that instead

**Reply by ensconced** [reply]
Thank you, however this is just a simplified example of my actual code. For the purposes of this question can we assume that the useEffect approach is necessary?

**Reply by EskiMojo14** [reply]
i still wouldn't recommend it, but anyway:

based on the [source code](https://github.com/reduxjs/redux-toolkit/blob/master/packages/toolkit/src/query/react/buildHooks.ts#L866), the trigger should only change reference when either `store.dispatch` or `api.endpoints[yourQuery].initiate` change, and both of those should be stable within the lifetime of your app.(HMR not withstanding)
